### PR TITLE
chore: Add documentation for Headings breaking changes for Screen Reader

### LINKED ISSUE #12731: Add documentation for Headings breaking changes for Screen Reader support
- [x] Document all the breaking changes 
- [x] List all affected pages and components

### PR DESCRIPTION
**Pull request by saad-mo**
Closes #12731 

@developpeurweb - can you pls review the documentation. Do we also need to document the changes made to the `scss` files? 

@Pucek9 - Can you pls review the documentation for `PageTitleComponent` and see if we need to add/remove any for this new component?

### CODE REVIEW

File: docs/migration/4_0.md [hunk 1]
```diff
@@ -783,6 +783,24 @@ The following properties where removed in 4.0:
 - Styles for following selectors `cx-close-account-modal`, `cx-close-account`, `cx-forgot-password`, `cx-register`, `cx-update-password-form`, `cx-user-form` were moved from `@spartacus/styles` to `@spartacus/user/profile/styles`.
 - Styles for following selectors `cx-login`, `cx-login-form` were moved from `@spartacus/styles` to `@spartacus/user/account/styles`.
 
+#### (a11y) Screen Reader Improvements: Headings
+
+- Most of these changes affect only headings on screen reader rotor feature. So changes are visible for sighted users. 
+- Store Names on Store finder list page changed to `h2` from a `div`
```
Comment (Pucek9) [inline docs/migration/4_0.md:789]: IMO we have separated part in documentation for component changes (like template), and fact that it's for a11y is less important than specification which component we edited. Please check other examples (e.g. CartListItemComponent)

I would see it more like
```
### StoreFinderListItemComponent
- `div[class.cx-store-name]` element has been changed to `h2[class.cx-store-name]` (a11y)

### StoreFinderStoresCountComponent
- ...
and so on
```

**Review by Pucek9 — CHANGES_REQUESTED** [review]
IMO we have separated part in documentation for component changes (like template), and fact that it's for a11y is less important than specification which component we edited. Please check other examples (e.g. `CartListItemComponent`)

I would see it more like
```
### StoreFinderListItemComponent
- `div[class.cx-store-name]` element has been changed to `h2[class.cx-store-name]` due to a11y.

### StoreFinderStoresCountComponent
- ...
```

and so on

**Comment by saad-mo** [comment]
> IMO we have separated part in documentation for component changes (like template), and fact that it's for a11y is less important than specification which component we edited. Please check other examples (e.g. `CartListItemComponent`)
> 
> I would see it more like
> 
> ```
> ### StoreFinderListItemComponent
> - `div[class.cx-store-name]` element has been changed to `h2[class.cx-store-name]` due to a11y.
> 
> ### StoreFinderStoresCountComponent
> - ...
> ```
> 
> and so on

Updated as requested.

File: docs/migration/4_0.md [hunk 2]
```diff
@@ -783,6 +783,61 @@ The following properties where removed in 4.0:
...
 - Styles for following selectors `cx-login`, `cx-login-form` were moved from `@spartacus/styles` to `@spartacus/user/account/styles`.
 
+### (a11y) Screen Reader Improvements: Headings
+- Most of these changes affect only headings on screen reader rotor feature. So changes are visible for sighted users. 
+#### StoreFinderListItemComponent
+- `div[class.cx-store-name]` element has been changed to `h2[class.cx-store-name]`.
+#### StoreFinderStoresCountComponent
+- `div[class.cx-title` element has been changed to `h2[class.cx-title]`
+
+#### div[class.cx-total]` element has been chanted to `h4[class.cx-total]`
```
Comment (Pucek9) [inline docs/migration/4_0.md]:
```suggestion
- Most of these changes affect only headings on screen reader rotor feature. So changes are visible for sighted users. 

#### StoreFinderListItemComponent
- `div[class.cx-store-name]` element has been changed to `h2[class.cx-store-name]`.

#### StoreFinderStoresCountComponent
- `div[class.cx-title]` element has been changed to `h2[class.cx-title]`
- `div[class.cx-total]` element has been chanted to `h4[class.cx-total]`
```

File: docs/migration/4_0.md [hunk 3]
```diff
@@ -783,6 +783,61 @@ The following properties where removed in 4.0:
...
+- `<h4>{{item.name}}</h4>` element changed to `<h3>{{item.name}}</h3>`
+
+#### WishListItemComponent 
+- `<a>{{ cartEntry.product.name }}</a>` element changed to `<a><h2>{{ cartEntry.product.name }}</h2></a>`
+
+#### CardComponent
+- `h4[class.cx-card-title]` element changed to `h3[class.cx-card-title]`
+
+#### CarouselComponent
+- `<h3 *ngIf="title">{{ title }}</h3>` element changed to `<h2 *ngIf="title">{{ title }}</h2>`
```
Comment (Pucek9) [inline docs/migration/4_0.md:837]: I would like to see it below other `component` change and above `###Translations (i18n) changes`

Comment (Pucek9) [inline docs/migration/4_0.md]: I would like to see it below other `component` changes and above `###Translations (i18n) changes`

File: docs/migration/css-changes-in-version-4.md [hunk 4]
```diff
@@ -87,3 +87,49 @@ title: Changes to Styles in 4.0
...
+* `h3` changed to `h2` to account for the change in markup template for improved screen reader support (a11y)
+
+## Product Carousel Component
+* `h4` changed to `h3` to account for the change in markup template for improved screen reader support (a11y)
+
+## Product List Item Component 
+* `h2` added to account for the change in markup template for improved screen reader support (a11y)
+
+## Wish List Item Component 
+* `h2` added to account for the change in markup template for improved screen reader support (a11y)
```
Comment (Pucek9) [inline docs/migration/css-changes-in-version-4.md]: are they really css changes?

Comment (saad-mo) [reply docs/migration/css-changes-in-version-4.md]: yes `h2` was added to the `scss` file

File: docs/migration/4_0.md [hunk 5]
```diff
@@ -719,6 +719,64 @@ The following properties where removed in 4.0:
 - Change detection strategy for this component was set to `OnPush`.
 - Spinner component was added to the template which relys on `isUpdating$` property.
 
+### LogoutGuard
+
```
Comment (Pucek9) [inline docs/migration/4_0.md:723]:
```suggestion
```

File: docs/migration/4_0.md [hunk 6]
```diff
@@ -719,6 +719,64 @@ The following properties where removed in 4.0:
...
+`AuthRedirectService` is a new, required constructor dependency.
```
Comment (Pucek9) [inline docs/migration/4_0.md]:
```suggestion
### LogoutGuard
`AuthRedirectService` is a new, required constructor dependency.
```

Comment (Pucek9) [reply docs/migration/4_0.md]: btw this is guard changes, it shouldn't be here. Please move it to other section

File: docs/migration/4_0.md [hunk 7]
```diff
@@ -719,6 +719,64 @@ The following properties where removed in 4.0:
...
+
+### (a11y) Screen Reader Improvements: Headings
+- Most of these changes affect only headings on screen reader elements dialog.
```
Comment (Pucek9) [inline docs/migration/4_0.md]: it shouldn't be a seperated category. Changes below are just component changes. This header can be removed

File: docs/migration/4_0.md [hunk 8]
```diff
@@ -719,6 +719,64 @@ The following properties where removed in 4.0:
...
+#### StoreFinderListItemComponent
+- `div[class.cx-store-name]` element has been changed to `h2[class.cx-store-name]`.
+#### StoreFinderStoresCountComponent
+- `div[class.cx-title` element has been changed to `h2[class.cx-title]`
```
Comment (Pucek9) [inline docs/migration/4_0.md]:
```suggestion
- `div[class.cx-store-name]` element has been changed to `h2[class.cx-store-name]`.

#### StoreFinderStoresCountComponent
- `div[class.cx-title]` element has been changed to `h2[class.cx-title]`
```

File: docs/migration/4_0.md [hunk 9]
```diff
@@ -719,6 +719,64 @@ The following properties where removed in 4.0:
...
+- `div[class.cx-review-cart-total]` element changed to `h4[class.cx-review-cart-total]`
+
+#### ShippingAddressComponent
+- `h3[class.cx-checkout-title]` element changed to `h2[class.cx-checkout-title]`
+
+#### BreadcrumbComponent
+- `isNotNullable`, `filter` import was removed 
+- `PageTitleComponent` import was added
+- `BreadcrumbComponent` extends `PageTitleComponent`
+- `setTitle()` function was moved to `PageTitleComponent`
```
Comment (Pucek9) [inline docs/migration/4_0.md]: We don't document stuff like import changes in ts files. 
```suggestion
#### BreadcrumbComponent
- `BreadcrumbComponent` extends `PageTitleComponent` now.
- `setTitle()` function has been moved to `PageTitleComponent`
```

File: docs/migration/4_0.md [hunk 10]
```diff
@@ -782,7 +840,3 @@ The following properties where removed in 4.0:
 
 - Styles for following selectors `cx-close-account-modal`, `cx-close-account`, `cx-forgot-password`, `cx-register`, `cx-update-password-form`, `cx-user-form` were moved from `@spartacus/styles` to `@spartacus/user/profile/styles`.
 - Styles for following selectors `cx-login`, `cx-login-form` were moved from `@spartacus/styles` to `@spartacus/user/account/styles`.
-
```
Comment (Pucek9) [inline docs/migration/4_0.md]: Please revert this change.

File: docs/migration/4_0.md (hunk 6 above)
Comment (Pucek9) [reply docs/migration/4_0.md]: yup, it looks like mistake -> https://github.com/SAP/spartacus/pull/12737/files#r651275286

File: docs/migration/4_0.md [hunk 11]
```diff
@@ -719,6 +719,80 @@ The following properties where removed in 4.0:
 - Change detection strategy for this component was set to `OnPush`.
 - Spinner component was added to the template which relys on `isUpdating$` property.
 
+#### StoreFinderListItemComponent
+
+- `div[class.cx-store-name]` element has been changed to `h2[class.cx-store-name]`.
+#### StoreFinderStoresCountComponent
```
Comment (Pucek9) [inline docs/migration/4_0.md]:
```suggestion
- `div[class.cx-store-name]` element has been changed to `h2[class.cx-store-name]`.

#### StoreFinderStoresCountComponent
```

File: docs/migration/css-changes-in-version-4.md [hunk 12]
```diff
@@ -87,3 +87,46 @@ title: Changes to Styles in 4.0
...
+
+## Checkout Media Style Component 
+* `type(3)` and `font-weight` added to retain existing styling after change in markup template for improved screen reader support (a11y)
+
+## `_index.scss` Changes 
+* `_screen-reader.scss` added which will contain screen reader specific styles (a11y)
+
+## `_screen-reader.scss` Changes
```
Comment (Pucek9) [inline docs/migration/css-changes-in-version-4.md]:
```suggestion
## Changes in Card Component 
* `.cx-card-title` class added (a11y)

## Changes in Carousel Component
* `h3` changed to `h2` to account for the change in markup template for improved screen reader support (a11y)

## Changes in Product Carousel Component
* `h4` changed to `h3` to account for the change in markup template for improved screen reader support (a11y)

## Changes in Product List Item Component 
* `h2` added to account for the change in markup template for improved screen reader support (a11y)

## Changes in Wish List Item Component 
* `h2` added to account for the change in markup template for improved screen reader support (a11y)

## Changes in Checkout Media Style Component 
* `type(3)` and `font-weight` added to retain existing styling after change in markup template for improved screen reader support (a11y)

## Changes in `_index.scss` Changes 
* `_screen-reader.scss` added which will contain screen reader specific styles (a11y)

## Changes in `_screen-reader.scss` Changes 
```

**Review by developpeurweb — APPROVED** [review]
All CSS changes for Headings seem to be now reflected in docs. Therefore approving that part.

File: docs/migration/4_0.md (hunk 10 above)
Comment (saad-mo) [reply docs/migration/4_0.md]: done

File: docs/migration/css-changes-in-version-4.md (hunk 4 above)
Comment (saad-mo) [reply docs/migration/css-changes-in-version-4.md]: discussed over the phone and resolved.

File: docs/migration/4_0.md (hunk 3 above)
Comment (saad-mo) [reply docs/migration/4_0.md]: done

File: docs/migration/4_0.md (hunk 2 above)
Comment (saad-mo) [reply docs/migration/4_0.md]: done

**Review by Pucek9 — APPROVED** [review]
